keps/sig-auth/20190116-service-account-external-signing.md

```diff
@@ -127,9 +159,43 @@ New token generation and validation could suffer a performance difference when r
 
 Signing and verifying tokens over a grpc API carries the risk of a server side request forgery, where a malicious client could generate tokens. To mitigate this risk, the API will only be accessible over a unix socket.
 
-## Graduation Criteria
+## Design Details
+
+### Test Plan
+
+Unit tests covering:
+* ProjectedToken signing works with external signer client
+* OIDC keys URL shows public keys from external signer
+
+Integration tests covering:
```

Comment:
Some thoughts:

- Tests that prove the API server keeps working for verification when the external signer is down (this does make me wonder if the public keys need a TTL)
- Tests to confirm that signing fails when the external signer is down

Reply:
I think that some may prefer a more decoupled approach where kms-plugin may cache the signing key that it received from a Secret Store (ex. Vault).

Reply:
Even if the kms-plugin cached the signing key, if its "down" from the perspective of the API server, signing operations would fail right?

Reply:
Right.
In my mind, "external signer" == remote KMS or Secrets Store, and kms-plugin is the intermediary between the kube-apiserver/controller-manager and such remote signers.
Therefore, external signers may fail (after kms-plugin cached the signing key, assuming that is the chosen architecture) without impacting the kube-apiserver (at least for the TTL of the cache entry).  
Having said that, (from the testing plan perspective) I don't think that this distinction matters since the choice of architecture is opaque to the api-server.